src/gatsby-node.js

```diff
@@ -302,6 +304,44 @@ exports.sourceNodes = async (
         );
       }
     };
+    ws.onclose = function() {
+      reconnectingWebSocket(ws, previousDelay);
+    };
+    ws.onerror = function(err) {
+      console.log(err.stack);
+    };
+
+    ws.on('pong', () => {
+      count--;
+    });
+
+    const intervalId = setInterval(() => {
+      ws.ping('heartBeat');
+      if (count == 2) {
+        reconnectingWebSocket(ws, previousDelay);
+      }
+      count++;
+    }, 60000);
+  };
+
+  const reconnectingWebSocket = function(ws, previousDelay) {
```

Comment:
This is confusing that there is `const previousDelay` which is later replaced with mutable argument.

Reply:
I changed the declaration from const to let to avoid the confusion. sorry!